lib/puppet/indirector/report/json.rb

```diff
@@ -25,7 +25,7 @@ def save(request)
 
     begin
       Puppet::FileSystem.replace_file(filename, mode) do |fh|
-        fh.print JSON.dump(request.instance)
+        fh.print JSON.pretty_generate(request.instance).force_encoding(Encoding::BINARY)
```

Comment:
Why is encoding forced to binary?

I prefer using the non-pretty `dump` for code that could execute in puppetserver for every agent request. If someone wants a pretty version, I'd recommend using `jq`. For example https://tickets.puppetlabs.com/browse/PUP-10608